catalog/data_volume.go

```diff
@@ -0,0 +1,50 @@
+package catalog
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/databricks/databricks-sdk-go"
+	"github.com/databricks/databricks-sdk-go/service/catalog"
+	"github.com/databricks/terraform-provider-databricks/common"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+type volumeDataParams struct {
+	FullName    string `json:"full_name,omitempty"`
+	CatalogName string `json:"catalog_name,omitempty"`
+	SchemaName  string `json:"schema_name,omitempty"`
+	Name        string `json:"name,omitempty"`
+}
+
+func (volumeDataParams) Aliases() map[string]string {
+	return map[string]string{"Id": "FullName"}
+}
+func (volumeDataParams) CustomizeSchema(s map[string]*schema.Schema) map[string]*schema.Schema {
+	common.CustomizeSchemaPath(s, "full_name").SetExactlyOneOf([]string{"catalog_name"}).SetComputed()
+	common.CustomizeSchemaPath(s, "catalog_name").SetRequiredWith([]string{"schema_name", "name"}).SetComputed()
+	common.CustomizeSchemaPath(s, "schema_name").SetRequiredWith([]string{"catalog_name", "name"}).SetComputed()
+	common.CustomizeSchemaPath(s, "name").SetRequiredWith([]string{"catalog_name", "schema_name"}).SetComputed()
```

Comment:
The interface should be `s.SchemaPath("name).SetRequiredWith`. This might be the reason why we are having the unit test failures